### PR TITLE
[DOXIA-728] Support blockquotes in MarkdownSink

### PR DESCRIPTION
**Pull request by kwin**
Streamline nested block handling
Always collapse subsequent newlines and blank lines

### CODE REVIEW

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java [hunk 1]
```diff
@@ -77,11 +77,11 @@ public interface MarkdownMarkup extends TextMarkup {
...
 
-    /** Syntax for the ordered list item: '1.' */
-    String LIST_ORDERED_ITEM_START_MARKUP = "1.";
+    /** Syntax for the ordered list item: '1. ' */
+    String LIST_ORDERED_ITEM_START_MARKUP = "1. ";
 
-    /** Syntax for the unordered list item: "-" */
-    String LIST_UNORDERED_ITEM_START_MARKUP = "-";
+    /** Syntax for the unordered list item: "- " */
+    String LIST_UNORDERED_ITEM_START_MARKUP = "- ";
```
Comment (michael-o) [inline doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java:84]: This is rather a fix, right?

**Review by michael-o — APPROVED** [review]
I don't have any objections, I trust your work here.

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java (hunk 1 above)
Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java:84]: Previously this was combined each time with a space, but it never occurs without it, so it makes sense to include the space in the constant.

**Comment by kwin** [comment]
There is currently an issue on Windows I still need to investigate.

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java (hunk 1 above)
Comment (michael-o) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java:84]: Makes sense.

**Comment by jira-importer** [comment]
Resolve #854

**Pull request merged**